File: tiq-e/dynamic-programming/1/main.ts
import {performance} from "perf_hooks";

function climbStairs(n: number): number {
  return Array(n - 1)
    .fill(0) 
    .reduce(([a, b]) => [b, a + b], [1, 1])
    .pop();
}

const n = 5;

const start = performance.now();
const result = climbStairs(n);
const end = performance.now();
console.log("Duration:", end - start);
console.log("Result:", result);

File: tiq-e/dynamic-programming/1/main.go
package main

import (
	"fmt"
	"time"
)

func climbStairs(n int) int {
	if n <= 2 {
		return n
	}
	a, b := 1, 2
	for i := 3; i < n; i++ {
		tmp := b
		b = b + a
		a = tmp
	}
	return a + b
}

func main() {
	var n = 5

	var start = time.Now()
	var result = climbStairs(n)
	var end = time.Since(start)
	fmt.Println("Duration: ", end)
	fmt.Println("Result: ", result)
}
